Sources/TuistCore/Models/Project.swift

```diff
@@ -28,6 +14,10 @@ public struct Project: Hashable, Equatable, CustomStringConvertible, CustomDebug
     /// Path to the Xcode project that will be generated.
     public var xcodeProjPath: AbsolutePath
 
+    /// If any of the targets of the project has a dependency with CocoaPods,
+    /// these paths points to the Xcode projects so that they get added to the generated workspace.
+    public var podProjectPaths: Set<AbsolutePath>
```

Comment:
I think in a follow-up PR we could deprecate the CocoaPods dependency and rather have something along the lines of:

```swift
import ProjectDescription

let project = Project(pods: ".")
```

Or even better, assume that the Pods for a project would be defined in a `Podfile` that lives in the same directory as the Project.swift. That way the interface could be:

```swift
import ProjectDescription

let project = Project(pods: true)
```

Thoughts @tuist/core ?

Reply:
Would it be safe to make an assumption that if a Podfile exists alongside your project manifest that it should contain cocoapods?

The boolean seems like a good choice here

Reply:
for most cases that is true, but pod files can live in a top level directory too :/

e.g.

```
MyApp
  - Project.swift
FrameworkA
  - Project.Swift
Podfile
Workspace.swift
```

If I'm not mistaken, there's usually 1 pod file for a workspace as opposed to per project, perhaps we can change the way we interact with CocoaPods such that it's no longer per project, but rather a workspace level or even post generation option to run `pod install`. This may help resolve https://github.com/tuist/tuist/issues/1226


CocoaPods allows defining paths to projects, as such it can simplify integration with Tuist

https://guides.cocoapods.org/syntax/podfile.html#project

e.g.

```rb
workspace 'MyWorkspace'

target 'MyApp' do
  project 'MyApp/MyApp'
  ...
end

target 'FrameworkA' do
  project 'FrameworkA/FrameworkA'
  ...
end
```